Firmware/SwitchPro/Joystick.c

```diff
@@ -0,0 +1,420 @@
+        // SET(LED_PORT, DON_LED_PIN);
+        // SET(LED_PORT, KAT_LED_PIN);
+        // Clear structs
+        for(int i = 0; i < KB_SWITCHES; i++) {
+            switches[i].state = 0;
+            if(switches[i].state != switches[i].lastReport) {
+                switchesChanged = 1;
+            }
+        }
+    }
+}
+
+void nunchuck_init(void) {
+    // try to say hello
+    if(!i2c_start(NUNCHUCK_ADDR | I2C_WRITE)) {
+        i2c_write(0xF0);
+        i2c_write(0x55);
+        i2c_stop();
+        _delay_ms(25);
+        
+        i2c_start(NUNCHUCK_ADDR | I2C_WRITE);
+        i2c_write(0xFB);
+        i2c_write(0x00);
+        i2c_stop();
+        _delay_ms(25);
+        nunchuck_online();
+    } else {
+        nunchuck_offline();
+    }
+}
+
+uint8_t nunchuck_readByte(uint8_t address) {
+    uint8_t data = 0xFF;
+    
+    if(!nunchuckReady) {
+        nunchuck_init();
+    }
+
+    if(!i2c_start(NUNCHUCK_ADDR | I2C_WRITE)) {
+        i2c_write(address);
+        i2c_stop();
+
+        i2c_start(NUNCHUCK_ADDR | I2C_READ);
+        data = i2c_readNak();
+        i2c_stop();
+        nunchuck_online();
+    } else {
+        nunchuck_offline();
+    }
+    return data;
+}
+
+// Starting at address, read n bytes and return the last
+void nunchuck_readMany(uint8_t address, uint8_t *data, uint8_t count) {
+    
+    if(!nunchuckReady) {
+        nunchuck_init();
+    }
+
+    if(!i2c_start(NUNCHUCK_ADDR | I2C_WRITE)) {
+        i2c_write(address);
+        i2c_stop();
+
+        i2c_start(NUNCHUCK_ADDR | I2C_READ);
+        for(uint8_t i = 0; i < count-1; i++) {
+            data[i] = i2c_readAck();
+        }
+        data[count-1] = i2c_readNak();
+        i2c_stop();
+        nunchuck_online();
+    } else {
+        nunchuck_offline();
+    }
+}
+
+typedef enum {
+    SYNC_CONTROLLER,
+    TATACON_PASSTHROUGH
+} State_t;
+
+State_t state = SYNC_CONTROLLER;
+#define ECHOES 2
+int echoes = 0;
+int report_count = 0;
+USB_JoystickReport_Input_t last_report;
+
+void GetNextReport(USB_JoystickReport_Input_t* const reportData) {
+    uint8_t data, i;
+    // Prepare an empty report
+    memset(reportData, 0, sizeof(USB_JoystickReport_Input_t));
+    reportData->LX = STICK_CENTER;
+    reportData->LY = STICK_CENTER;
+    reportData->RX = STICK_CENTER;
+    reportData->RY = STICK_CENTER;
+    reportData->HAT = HAT_CENTER;
+
+    // Repeat ECHOES times the last report
+    if (echoes > 0) {
+        memcpy(reportData, &last_report, sizeof(USB_JoystickReport_Input_t));
+        echoes--;
+        return;
+    }
+
+    // States and moves management
+    switch (state) {
+        case SYNC_CONTROLLER:
+            if (report_count > 100) {
+                report_count = 0;
+                state = TATACON_PASSTHROUGH;
+            }
+            else if (report_count == 25 || report_count == 50) {
+                reportData->Button |= SWITCH_L | SWITCH_R;
+            }
+            else if (report_count == 75 || report_count == 100) {
```

Comment:
A bit too quick for the switch to catch the A button :)
I set my second value to 200 instead of 100, seems to do the trick